packages/material-ui-lab/src/Autocomplete/Autocomplete.test.js

```diff
@@ -1,33 +1,207 @@
+
+      const popup = getByRole('listbox');
+      expect(combobox, 'combobox owns listbox').to.have.attribute(
+        'aria-owns',
+        popup.getAttribute('id'),
+      );
+      expect(textbox).to.have.attribute('aria-controls', popup.getAttribute('id'));
+      expect(textbox, 'no option is focused when openened').not.to.have.attribute(
+        'aria-activedescendant',
+      );
+
+      const options = getAllByRole('option');
+      expect(options).to.have.length(2);
+      options.forEach(option => {
+        expect(popup).to.contain(option);
+      });
+
+      const buttons = getAllByRole('button');
+      expect(buttons).to.have.length(2);
+      // TODO: computeAccessibleName
+      expect(buttons[0]).to.have.attribute('title', 'Clear');
+      // TODO: computeAccessibleName
+      expect(buttons[1]).to.have.attribute('title', 'Close popup');
+      buttons.forEach(button => {
+        expect(button, 'button is not in tab order').to.have.property('tabIndex', -1);
+      });
+    });
+  });
+
+  describe('when popup closed', () => {
+    it('opens when the textbox is focused', () => {
+      const handleOpen = spy();
+      render(
+        <Autocomplete
+          onOpen={handleOpen}
+          renderInput={params => <TextField autoFocus {...params} />}
+        />,
+      );
+
+      expect(handleOpen.callCount).to.equal(1);
+    });
+
+    ['ArrowDown', 'ArrowUp'].forEach(key => {
+      it(`opens on ${key} when focus is on the textbox without moving focus`, () => {
+        const handleOpen = spy();
+        const { getByRole } = render(
+          <Autocomplete
+            open={false}
+            onOpen={handleOpen}
+            renderInput={params => <TextField autoFocus {...params} />}
+          />,
+        );
+
+        fireEvent.keyDown(document.activeElement, { key });
+
+        // first from focus
+        expect(handleOpen.callCount).to.equal(2);
+        expect(getByRole('textbox')).not.to.have.attribute('aria-activedescendant');
+      });
+    });
+
+    it('does not clear the textbox on Escape', () => {
+      const handleChange = spy();
+      render(
+        <Autocomplete
+          onChange={handleChange}
+          open={false}
+          options={['one', 'two']}
+          value="one"
+          renderInput={params => <TextField autoFocus {...params} />}
+        />,
+      );
+
+      fireEvent.keyDown(document.activeElement, { key: 'Escape' });
+
+      expect(handleChange.callCount).to.equal(0);
+    });
+  });
+
+  describe('when popup open', () => {
+    it('closes the popup if Escape is pressed ', () => {
+      const handleClose = spy();
+      render(
+        <Autocomplete
+          onClose={handleClose}
+          open
+          options={['one', 'two  ']}
+          renderInput={params => <TextField autoFocus {...params} />}
+        />,
+      );
+
+      fireEvent.keyDown(document.activeElement, { key: 'Escape' });
+
+      expect(handleClose.callCount).to.equal(1);
```

Comment:
This will be called twice with jsdom actual because the implementation includes a .focus() noop that causes jsdom to fire a blur event which also calls onClose. In the browser this will work correctly.